src/components/ConfirmPopup.js

```diff
@@ -0,0 +1,45 @@
+import Popup from "./Popup";
+
+class ConfirmPopup extends Popup {
+  constructor(selector) {
+    super(selector);
+    this._form = this._popupElement.querySelector(".modal__container");
+    this._submitButton = this._popupElement.querySelector(
+      ".modal__save-btn_delete"
+    );
+    this._submitbuttonTextContent = this._submitButton.textContent;
+  }
+
+  setEventListeners() {
+    super.setEventListeners();
+    this._popupElement
+      .querySelector("#confirm-delete-btn")
+      .addEventListener("click", () => {
+        if (this._submitFunction) {
+          this._submitFunction();
+        }
+      });
+  }
+
+  setSubmitFunction(submitFunction) {
+    this._submitFunction = submitFunction;
+  }
+
+  setIsDeleting(isDeleting) {
+    if (isDeleting) {
+      this._submitButton.textContent = "Deleting...";
+    } else {
+      this._submitButton.textContent = this._submitbuttonTextContent;
+    }
+  }
+
+  open() {
+    super.open();
+  }
+
+  close() {
+    super.close();
```

Comment:
You shouldn’t create the **same methods** of the parent class if you don’t change them inside and only call them with `super`. The parent methods are available in any inheritor of the class.